avalon/nuke/lib.py

```diff
@@ -3,82 +3,283 @@
 import nuke
 import re
 import logging
-import toml
+import six
+from ..vendor import clique
 
 log = logging.getLogger(__name__)
 
 
 @contextlib.contextmanager
 def maintained_selection():
+    """Maintain selection during context
+
+    Example:
+        >>> with maintained_selection():
+        ...     node['selected'].setValue(True)
+        >>> print(node['selected'].value())
+        False
+    """
+    nodes = nuke.allNodes()
     previous_selection = nuke.selectedNodes()
     try:
         yield
     finally:
         # unselect all selection in case there is some
-        current_seletion = nuke.selectedNodes()
-        [n['selected'].setValue(False) for n in current_seletion]
+        [n['selected'].setValue(False) for n in nodes]
         # and select all previously selected nodes
         if previous_selection:
-            [n['selected'].setValue(True) for n in previous_selection]
+            [n['selected'].setValue(True)
+             for n in nodes
+             if n.name() in previous_selection]
+
+
+def reset_selection():
+    """Deselect all selected nodes
+    """
+    nodes = nuke.allNodes()
+    [n['selected'].setValue(False) for n in nodes]
 
 
-def add_avalon_tab_knob(node):
-    """Adding a tab and a knob into a node
+def select_nodes(nodes):
+    """Selects all inputed nodes
+
+    Arguments:
+        nodes (list): nuke nodes to be selected
     """
+    assert isinstance(nodes, (list, tuple)), "nodes has to be list or tuple"
+
+    [n['selected'].setValue(True) for n in nodes]
+
+
+def add_publish_knob(node):
+    '''Adds Publish node to node
+
+    Arguments:
+        node (obj): nuke node to be processed
+
+    Returns:
+        node (obj): processed nuke node
+    '''
+    if "publish" not in node.knobs():
+        divider = nuke.Text_Knob('')
+        knob = nuke.Boolean_Knob("publish", "Publish", True)
+        knob.setFlag(0x1000)
+        node.addKnob(divider)
+        node.addKnob(knob)
+    return node
+
+
+def set_avalon_knob_data(node, data={}, prefix="ak:"):
```

Comment:
I'm not familiar with Nuke. But is there a reason you went for prefix `ak:` (Avalon Knob?) instead of `avalon:`? Is it required to be of a very short form. Why not be more explicit?

Reply:
no particular reason. It could be `avalon` or it could be whatever